### ISSUE TITLE
import tensorflow failed, "ImportError: DLL load failed". Even after install visual studio 2015, Microsoft Visual C++ 2015 Redistributable Update 3.

### ISSUE THREAD

**Issue by sdhhqb** [tensorflow/tensorflow#17393]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: win10 x64
- **TensorFlow installed from (source or binary)**:  pip install --upgrade tensorflow
- **TensorFlow version (use command below)**: 1.6.0
- **Python version**: 3.6.4
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

### Describe the problem
I was trying to install tensorflow **cpu version** on windows10, but always got error when import tensorflow.

I read the [common_installation_problems](https://www.tensorflow.org/install/install_windows#common_installation_problems) , tried many solution I found on github, stackoverflow, etc. I install visual studio 2015, visual studio 2017, [Microsoft Visual C++ 2015 Redistributable Update 3](https://www.microsoft.com/en-us/download/details.aspx?id=53587)(both 32 and 64),  msvcp140.dll can find in both System32 and SysWow64 folder. But still can't import tensorflow.

Is there something I missed out?

### Source code / logs
>import tensorflow error info:

```
Python 3.6.4 (v3.6.4:d48eceb, Dec 19 2017, 06:54:40) [MSC v.1900 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
Traceback (most recent call last):
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 658, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 571, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 922, in create_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
ImportError: DLL load failed: 动态链接库(DLL)初始化例程失败。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 21, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow_internal')
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ModuleNotFoundError: No module named '_pywrap_tensorflow_internal'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\__init__.py", line 24, in <module>
    from tensorflow.python import *
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\python\__init__.py", line 49, in <module>
    from tensorflow.python import pywrap_tensorflow
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 74, in <module>
    raise ImportError(msg)
ImportError: Traceback (most recent call last):
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 658, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 571, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 922, in create_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
ImportError: DLL load failed: 动态链接库(DLL)初始化例程失败。

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 21, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 20, in swig_import_helper
    return importlib.import_module('_pywrap_tensorflow_internal')
  File "C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ModuleNotFoundError: No module named '_pywrap_tensorflow_internal'


Failed to load the native TensorFlow runtime.

See https://www.tensorflow.org/install/install_sources#common_installation_problems

for some common reasons and solutions.  Include the entire stack trace
above this error message when asking for help.
```

>run tensorflow_self_check.py result:

```
ERROR: Failed to import the TensorFlow module.

- Python version is 3.6.

- TensorFlow is installed at: C:\Users\sss\AppData\Local\Programs\Python\Python36\lib\site-packages\tensorflow

- Could not load 'cudart64_80.dll'. The GPU version of TensorFlow
  requires that this DLL be installed in a directory that is named in
  your %PATH% environment variable. Download and install CUDA 8.0 from
  this URL: https://developer.nvidia.com/cuda-toolkit

- Could not load 'nvcuda.dll'. The GPU version of TensorFlow requires that
  this DLL be installed in a directory that is named in your %PATH%
  environment variable. Typically it is installed in 'C:\Windows\System32'.
  If it is not present, ensure that you have a CUDA-capable GPU with the
  correct driver installed.

- Could not load 'cudnn64_5.dll'. The GPU version of TensorFlow
  requires that this DLL be installed in a directory that is named in
  your %PATH% environment variable. Note that installing cuDNN is a
  separate step from installing CUDA, and it is often found in a
  different directory from the CUDA DLLs. You may install the
  necessary DLL by downloading cuDNN 5.1 from this URL:
  https://developer.nvidia.com/cudnn

- Could not find cuDNN.
```

**Comment by thuojb** [comment]
I'm also having the same issue. ANybody with a fix?

**Comment by thuojb** [comment]
Seems tensorflow 1.6 is the issue,
I installed 1.5 and it works just fine
try 
pip install tensorflow==1.5

**Comment by fo40225** [comment]
https://github.com/tensorflow/tensorflow/issues/17386#issuecomment-370129452

**Comment by gunan** [comment]
Closing, duplicate of #17386

**Comment by DanS2003** [comment]
@thuojb, thanks. it fixed my problem

**Comment by samuelprince77** [comment]
@thuojb thanks mate. This solution still works till this day running tensorflow 1.8. Life saver

**Comment by MDShahrouq** [comment]
@thuojb Thanks man, it still works.

But can you tell me how can we switch to tensorflow 1.8 without any errors while running

**Comment by XuanxuanGao** [comment]
@thuojb It really works! Thank you~

**Comment by Abdulr-intija** [comment]
Have in mind that downgrading to tensorflow 1.5 might not be the best solution because for instance, you won't have access to some of the new features of tensorflow like the 'eager execution' feature for example.

What i did in my case was to install tensorflow 1.8 with conda instead of pip. From my little experience with conda and pip, conda seems to perform better when it comes to placing required files in their proper location.

In summary, Instead of `pip3 install --upgrade tensorflow`, i used `conda install tensorflow `which will install the latest version properly.

In case you don't have Anaconda installed for conda commands. Download from [https://www.continuum.io/downloads](url)

**Comment by IverBand** [comment]
@Abdulr-intija conda install tensorflow worked for me in an identical situation.  Many thanks for ending my struggle!

**Comment by katamit** [comment]
@thuojb,  thanks,  yes it works with version 1.5.0 
@Abdulr-intija **conda install tensorflow** worked for me too.  able to install 1.8.0 successfully. thanks

**Comment by ankit338841** [comment]
i am getting the same issue with tensorflow 1.8.0.
can anybody help me with this issue?

**Comment by Mdelaf** [comment]
Same issue here. It worked with version 1.5.

This problem could occur when the CPU lacks AVX instruction. [Using a wheel built without AVX also fix problem](https://github.com/tensorflow/tensorflow/issues/17386#issuecomment-370129452).

**Comment by Scientificer** [comment]
thuojb, thank you so much! I am so happy I got it working.

**Comment by bejo2018** [comment]
i have same problem, when i import show error like this

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
c:\python\lib\site-packages\tensorflow\python\pywrap_tensorflow.py in <module>
     57 
---> 58   from tensorflow.python.pywrap_tensorflow_internal import *
     59   from tensorflow.python.pywrap_tensorflow_internal import __version__

c:\python\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py in <module>
     27             return _mod
---> 28     _pywrap_tensorflow_internal = swig_import_helper()
     29     del swig_import_helper

c:\python\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py in swig_import_helper()
     23             try:
---> 24                 _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
     25             finally:

c:\python\lib\imp.py in load_module(name, file, filename, details)
    241         else:
--> 242             return load_dynamic(name, filename, file)
    243     elif type_ == PKG_DIRECTORY:

c:\python\lib\imp.py in load_dynamic(name, path, file)
    341             name=name, loader=loader, origin=path)
--> 342         return _load(spec)
    343 

ImportError: DLL load failed with error code -1073741795

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
<ipython-input-1-64156d691fe5> in <module>
----> 1 import tensorflow as tf

c:\python\lib\site-packages\tensorflow\__init__.py in <module>
     20 
     21 # pylint: disable=g-bad-import-order
---> 22 from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
     23 
     24 try:

c:\python\lib\site-packages\tensorflow\python\__init__.py in <module>
     47 import numpy as np
     48 
---> 49 from tensorflow.python import pywrap_tensorflow
     50 
     51 # Protocol buffers

c:\python\lib\site-packages\tensorflow\python\pywrap_tensorflow.py in <module>
     72 for some common reasons and solutions.  Include the entire stack trace
     73 above this error message when asking for help.""" % traceback.format_exc()
---> 74   raise ImportError(msg)
     75 
     76 # pylint: enable=wildcard-import,g-import-not-at-top,unused-import,line-too-long

ImportError: Traceback (most recent call last):
  File "c:\python\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "c:\python\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "c:\python\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
  File "c:\python\lib\imp.py", line 242, in load_module
    return load_dynamic(name, filename, file)
  File "c:\python\lib\imp.py", line 342, in load_dynamic
    return _load(spec)
ImportError: DLL load failed with error code -1073741795

my sistem are:
- windows 7 64 bit
- python 3.5.2rc1
- intel core 2 due e8400

**Comment by darkmanAmazigh** [comment]
i got this message with tenserflow 1.10.0,
solution ?

**Comment by iDiniy** [comment]
@Abdulr-intija   thank you,i finaly resolve it .thank you

**Comment by testlump** [comment]
For reference, using ProcessMonitor is a good way to determine which DLL can't be loaded by tensorflow. Documented here: http://shitalshah.com/p/debugging-tensorflow-dll-importerror/

**Comment by na6an** [comment]
Win 10 64bit. python 3.5.2, CUDA 8.0, cuDNN v5.1 still having DLL load failed on both tensorflow-gpu 1.10 and 1.12. Other setups tested including python 3.6.7, CUDA 9.0/10.0, cuDNN 7.4.1 for 10.0, 7.4.1/7.3.1 for 9.0.  Confirmed msvcp140.dll, added cuda/bin, include, lib paths, made a copy for both cudnn64_5.dll and  cudnn64_6.dll, reinstalled python with Tcl/Tk, etc. Nothing worked so far.

I even tried ProcessMonitor @testlump posted above, and last two dlls were "C:\Windows\System32\nvcuda.dll" and "C:\Windows\System32\setupapi.dll" However, it didn't fail. Does anyone have a clue?

**Comment by testlump** [comment]
You might find it's best to recompile from source. This resolved the issue for me (see below for more information).

Here are some things I found compiling on Windows 10 with GPU support: 

- Check your CPU instruction set supports AVX with [CPU-Z](https://www.cpuid.com/softwares/cpu-z.html) or an equivalent tool. If this isn't present (E.g. you're running an older CPU), you'll likely need to recompile tensorflow from source if you want it to work (this will likely apply irrespective of the OS). 

- Check you've got the correct CUDA libraries installed. The release Python package version of tensorflow (1.12) is built against `CUDA 9.0` (I used package: `cuda_9.0.176_win10.exe`) and CuDNN 7.0 (I used package: `cudnn-9.0-windows10-x64-v7.4.1.5.zip`). If you have multiple CUDA versions installed side-by-side (v7.0, v9.0, v10.0) etc which you are not using, uninstall them. 

- Make sure your CUDA *System* environment variables are configured (E.g. for v9.0): 
                - Variable: `CUDA_PATH` Value: `C:\Program Files\NVIDIA CPU Computing Toolkit\CUDA\v9.0`
                - Variable: `CUDA_PATH_V9_0` Value: `C:\Program Files\NVIDIA CPU Computing Toolkit\CUDA\v9.0`
 
- The CuDNN libraries should be unzipped to the CUDA install path, on Windows by default this is `C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v9.0 for version 9.0 `

**Building from source**

- Install `BAZEL` and `MSYS2` to their default locations (recommended). E.g. `C:\MSYS2\` and `C:\Bazel\`

- If you want to build the Python package from source follow these instructions: [https://www.tensorflow.org/install/source_windows](https://www.tensorflow.org/install/source_windows). Ensure you configure your `BAZEL` and `MSYS2` installations according to their respective web-pages (linked from tensorflow), pay particular attention to environment variables. If you've installed CUDA, CuDNN you can skip the section **Install GPU support (optional)**

- Clone or download the source as per the instructions and expand (E.g. `C:\temp\tensorflow-master\`)

- Edit the source file `{sourcegit for tensorflow path}\third_party\gpus\cuda_configure.bzl` and change the line `["grep", "--color=never", "-A1", "-E", define,` to: `["C:/msys64/usr/bin/grep", "--color=never", "-A1", "-E", define,`  (RESPECT THE INDENTATION!)

- If you have a username with a space in it - for example `Peter Blakey` so you users folder is `C:\users\Peter Blakey\`, create a symlink from 'Peter' in C:\Users\ - E.g. `C:\users\Peter -> C:\users\Peter Blakey\` - The white-space causes path resolution problems during the build. E.g. from a command prompt in `C:\users\` type `mklink /d Peter "C:\Users\Peter Blakey"` - you should now be able to `cd` in `c:\users\Peter\` and see the contents of `c:\users\Peter Blakey`

- If you have a username with a space in it, edit your *User* environment variables to path to `Python.exe` at the symlinked directory. E.g. Edit `c:\users\Peter Blakey\AppData\Local\Programs\Python\Python36\` to `C:\users\Peter\AppData\Local\Programs\Python\Python36\` and  `c:\users\Peter Blakey\AppData\Local\Programs\Python\Python36\Scripts\` to `c:\users\Peter Blakey\AppData\Local\Programs\Python\Python36\Scripts\`

- Create a empty folder for the build - E.g. `C:\temp\buildcache` - make sure you have at least 20GB free.


- When you execute the prebuild step `python configure.py` from the location of the source you cloned from github: 
      - Check the python path or override *No whitespace should be in the pathname*.
      - Check the python packages path or override *No whitespace should be in the pathname*.
      - Choose N for XLA JIT support 
      - Choose N for ROCm support
      - Choose Y for CUDA support
      - Enter the version of CUDA you want to use or accept the default (9.0)
      - Check the CUDA library path exists or if installed to another location, override.
      - Enter the version of cuDNN you want to use or accept the default (7)
      - Check the cuDNN path exists or if installed to another location, override
      - When prompted for the compute capability you want to enable for your GPU, visit [https://developer.nvidia.com/cuda-gpus](https://developer.nvidia.com/cuda-gpus) and enter the correct value. E.g. for a `Geforce GTX 970` it was 5.2 
       - Hit enter to accept the default optimization flags 
       - Choose `Yes` when asked to override eigen strong inline (Speeds up compilation time). 

**Build Step**

- Your compilation command should be something like this: `bazel --output_user_root=C:/temp/buildcache build --config=opt --config=cuda --define=no_tensorflow_py_deps=true //tensorflow/tools/pip_package:build_pip_package`

** or ** 

- If your CPU *does not* support AVX instruction set, when you build tensorflow omit the `--config=opt ` flag so you build with `bazel --output_user_root=C:/temp/buildcache build --config=cuda --define=no_tensorflow_py_deps=true //tensorflow/tools/pip_package:build_pip_package`

Once it's built (takes several hours), follow the instructions to build the pip package (E.g. execute `bazel-bin\tensorflow\tools\pip_package\build_pip_package C:/tmp/tensorflow_pkg` and then install the resulting Python wheel package to your environment).

Finally, test the installed tensorflow with: 

`python -c "import tensorflow as tf; tf.enable_eager_execution(); print(tf.reduce_sum(tf.random_normal([1000, 1000])))"`

**Other notes** 

- If the build fails, make sure you run `bazel clean` and delete the contents of the buildcache folder (E.g. `c:\temp\buildcache`) before commencing a new build

- You can build against the latest version of CUDA (v10.0) and cuDNN (v7) if you want. You may wish to try building against the supported version first, once you've verified that you can uninstall CUDA / CuDNN, install the latest versions and rebuild tensorflow.

**Comment by na6an** [comment]
@testlump I appreciate the detailed help. I actually attempted to compile from source first because the original pip install didn't work. Then, I soon realized fixing pip install could be easier than setting up the environment to build from source. Now it's circling back to source compiling again... I didn't want to run this on Ubuntu because the package dependency is pretty messed up there already and it keeps crashing, but perhaps I should try on Ubuntu. I will try source compiling as the last resort if that doesn't work either. Thanks for the help anyways...

**Comment by vinod270989verma** [comment]
Facing the same issue on my machine after some trials found a solution which worked for me.
Steps to solve the issue:
1. Re-Install Cuda(Till Date Tensorflow only supports Cuda V9.0 so install Cuda9.0 only)
2. Add Cuda library, bin paths to system variables
3. Add Cudnn(Any Version is Ohkey) library to Cuda9.0
4. Restart PC
5. Install "pip install tensorflow-gpu==1.8"(I tried with higher version also but not supported in my case but 1.8 easily got installed)

**Comment by Vini-S** [comment]
--------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
~\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py in <module>
     57 
---> 58   from tensorflow.python.pywrap_tensorflow_internal import *
     59   from tensorflow.python.pywrap_tensorflow_internal import __version__

~\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py in <module>
     27             return _mod
---> 28     _pywrap_tensorflow_internal = swig_import_helper()
     29     del swig_import_helper

~\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py in swig_import_helper()
     23             try:
---> 24                 _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
     25             finally:

~\Anaconda3\lib\imp.py in load_module(name, file, filename, details)
    241         else:
--> 242             return load_dynamic(name, filename, file)
    243     elif type_ == PKG_DIRECTORY:

~\Anaconda3\lib\imp.py in load_dynamic(name, path, file)
    341             name=name, loader=loader, origin=path)
--> 342         return _load(spec)
    343 

ImportError: DLL load failed with error code -1073741795

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
<ipython-input-10-491f9b70df53> in <module>
      1 import pandas as pd
----> 2 from keras.models import Sequential
      3 from keras.layers import Dense
      4 import numpy as np
      5 from sklearn.model_selection import train_test_split

~\Anaconda3\lib\site-packages\keras\__init__.py in <module>
      1 from __future__ import absolute_import
      2 
----> 3 from . import utils
      4 from . import activations
      5 from . import applications

~\Anaconda3\lib\site-packages\keras\utils\__init__.py in <module>
      4 from . import data_utils
      5 from . import io_utils
----> 6 from . import conv_utils
      7 
      8 # Globally-importable utils.

~\Anaconda3\lib\site-packages\keras\utils\conv_utils.py in <module>
      7 from six.moves import range
      8 import numpy as np
----> 9 from .. import backend as K
     10 
     11 

~\Anaconda3\lib\site-packages\keras\backend\__init__.py in <module>
     87 elif _BACKEND == 'tensorflow':
     88     sys.stderr.write('Using TensorFlow backend.\n')
---> 89     from .tensorflow_backend import *
     90 else:
     91     # Try and load external backend.

~\Anaconda3\lib\site-packages\keras\backend\tensorflow_backend.py in <module>
      3 from __future__ import print_function
      4 
----> 5 import tensorflow as tf
      6 from tensorflow.python.framework import ops as tf_ops
      7 from tensorflow.python.training import moving_averages

~\Anaconda3\lib\site-packages\tensorflow\__init__.py in <module>
     22 
     23 # pylint: disable=g-bad-import-order
---> 24 from tensorflow.python import pywrap_tensorflow  # pylint: disable=unused-import
     25 
     26 from tensorflow._api.v1 import app

~\Anaconda3\lib\site-packages\tensorflow\python\__init__.py in <module>
     47 import numpy as np
     48 
---> 49 from tensorflow.python import pywrap_tensorflow
     50 
     51 # Protocol buffers

~\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py in <module>
     72 for some common reasons and solutions.  Include the entire stack trace
     73 above this error message when asking for help.""" % traceback.format_exc()
---> 74   raise ImportError(msg)
     75 
     76 # pylint: enable=wildcard-import,g-import-not-at-top,unused-import,line-too-long

ImportError: Traceback (most recent call last):
  File "C:\Users\admin\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow.py", line 58, in <module>
    from tensorflow.python.pywrap_tensorflow_internal import *
  File "C:\Users\admin\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 28, in <module>
    _pywrap_tensorflow_internal = swig_import_helper()
  File "C:\Users\admin\Anaconda3\lib\site-packages\tensorflow\python\pywrap_tensorflow_internal.py", line 24, in swig_import_helper
    _mod = imp.load_module('_pywrap_tensorflow_internal', fp, pathname, description)
  File "C:\Users\admin\Anaconda3\lib\imp.py", line 242, in load_module
    return load_dynamic(name, filename, file)
  File "C:\Users\admin\Anaconda3\lib\imp.py", line 342, in load_dynamic
    return _load(spec)
ImportError: DLL load failed with error code -1073741795


Failed to load the native TensorFlow runtime.

See https://www.tensorflow.org/install/errors

for some common reasons and solutions.  Include the entire stack trace
above this error message when asking for help.

dataframe = pd.read_csv("dia

**Comment by isahhin** [comment]
Hi, I solved the problem as.

First, I created the conda environment as following:
**_conda create -n myenv  python=3.6
conda activate myenv_** 
Then I installed the keras-gpu=2.2.2 
**_conda install keras-gpu=2.2.2_** #this command also installed the tensorflow

**Comment by fengyoulin** [comment]
Open ...\Python37\Lib\site-packages\tensorflow\python\_pywrap_tensorflow_internal.pyd use [Dependency Walker](http://www.dependencywalker.com/), it will show you the DLL dependency tree, you will find which DLL cause the problem. TensorFlow always linked to the specific CUDA version.

**Comment by ajmalfaiz** [comment]
I fixed it.
Spend days searching to fix them.

Installed tensorflow with pip install tensorflow-cpu
Updated visual c++ 2015-2019

issue was because i didn't had nvidia graphics so that i should install with cpu version because i am using Raedon vega 8 graphics

Hope it helps.

**Comment by piyu872** [comment]
(base) PS C:\Users\Priyanshi> pip install tensorflow==1.5
ERROR: Could not find a version that satisfies the requirement tensorflow==1.5 (from versions: 1.13.0rc1, 1.13.0rc2, 1.13.1, 1.13.2, 1.14.0rc0, 1.14.0rc1, 1.14.0, 1.15.0rc0, 1.15.0rc1, 1.15.0rc2, 1.15.0rc3, 1.15.0, 1.15.2, 2.0.0a0, 2.0.0b0, 2.0.0b1, 2.0.0rc0, 2.0.0rc1, 2.0.0rc2, 2.0.0, 2.0.1, 2.1.0rc0, 2.1.0rc1, 2.1.0rc2, 2.1.0)
ERROR: No matching distribution found for tensorflow==1.5
i found this kind of error please anyone help me?

**Comment by payman-ap** [comment]
Simply downgrade your TF version!

pip uninstall tensorflow
pip install tensorflow==2.0.0

(for me happened when I upgraded to 2.1.0 and I had problem with any pip version)

**Comment by mihaimaruseac** [comment]
If it works on 2.0.0 but not on 2.1.0 then you need to install the 2019 MSVC redistributable as [the requirements state](https://www.tensorflow.org/install/pip)

If it doesn't work on 2.0.0 either, then check if your Python interpreter is on 64 bits (it has to be) and if your CPU supports AVX (it has to). If Python interpreter is not right, please install the right one. If CPU does not support AVX, then you can compile from source.

Since this is the solution, locking conversation to prevent bad advice and "it works for me" messages
